lib/ec2metadata/__init__.py

```diff
@@ -119,20 +109,47 @@ def _reset_meta_options_api_map(self):
         }
 
     def _set_api_header(self):
-        """Set the header to be used in requests to the metadata service,
-           IMDs. Prefer IMDSv2 which requires a token."""
+        """Set the header to be used in requests to the metadata service"""
         request = urllib.request.Request(
-            'http://169.254.169.254/latest/api/token',
+            'http://%s/latest/api/token' % self.addr,
             headers={'X-aws-ec2-metadata-token-ttl-seconds': '21600'},
             method='PUT'
         )
         try:
             token = urllib.request.urlopen(request).read().decode()
         except urllib.error.URLError:
-            self.request_header = {}
+            raise EC2MetadataError('Unable to retrieve metadata token')
 
         self.request_header = {'X-aws-ec2-metadata-token': token}
-        
+
+    def _set_ipaddress(self):
+        metadata_ip_addrs = {
+            'ipv6_addr': 'fd00:ec2::254',
+            'ipv4_addr': '169.254.169.254'
+        }
+        # Check if the Python implementation has IPv6 support in the first place
+        if not has_ipv6:
+            self.addr = metadata_ip_addrs.get('ipv4_addr')
```

Comment:
should we not exit here ?
if no exit, we will try IPv6 in the for loop on line 136
that will try IPv6 addr when we already know it has no IPv6

Reply:
something like
```python
metadata_ip_addrs = {}
if has_ipv6:
    metadata_ip_addrs['ipv6_addr'] = ...

metadata_ip_addrs['ipv4_addr'] = ...
```
that way we could preserve the order of IPv6 first and not try IPv6 if not included